packages/i18n/src/index.ts

```diff
@@ -1,29 +1,42 @@
 import * as _ from 'lodash';
-import { getTranslations, i18nReducer } from './reducers';
+import * as moment from 'moment';
+import { isPlainLabel, Labels } from '@jsonforms/core';
+import { getLocale, getTranslations, i18nReducer } from './reducers';
 
 export interface Translations {
   [key: string]: string;
 }
 
-export const translateLabel: (translations: Translations,
-                              label: string) => string =
-  (translations, label) => {
+const translate = (translations: Translations, label: string): string => {
 
-    if (translations && _.startsWith(label, '%')) {
-      const labelKey = label.substr(1, label.length);
+  if (translations && _.startsWith(label, '%')) {
+    const labelKey = label.substr(1, label.length);
 
-      return translations[labelKey] ? translations[labelKey] : label;
-    }
+    return translations[labelKey] ? translations[labelKey] : label;
+  }
+
+  return label;
+};
 
-    return label;
+export const translateLabel =
+  (translations: Translations, label: string | Labels): (string | Labels) => {
+
+    if (isPlainLabel(label)) {
+      return translate(translations, label);
+    } else {
+      return _.mapValues(label, l => translate(translations, l)) as Labels;
+    }
   };
 
 export const translateProps = (state, props) => {
   const label = translateLabel(getTranslations(state), props.label);
+  const localLocale = moment().locale(getLocale(state));
 
   return {
     ...props,
-    label
+    label,
+    locale: getLocale(state),
+    localLocale
```

Comment:
shoudn't we explicitly name this "momentLocale" ?

Reply:
Makes sense, update on its way.